src/App.tsx

```diff
@@ -20,23 +23,94 @@ import '@ionic/react/css/display.css';
 /* Theme variables */
 import './theme/variables.css';
 
-import Toolbar from './components/toolbar/toolbar';
-import SideMenu from './components/sideMenu/sideMenu';
-import Map from './components/map/map';
-import SearchBar from './components/searchbar/searchBar';
+import './theme/app.css';
+
+import Toolbar from './components/Toolbar';
+import SideMenu from './components/SideMenu';
+import Map from './components/Map';
+import SearchBar from './components/Searchbar';
+import Spinner from './components/UI/Spinner/Spinner';
+import { RootState } from './reducers';
+
+import { handleWindowSizeChange } from './service/general/checkScreenSize/checkScreenSize';
+// import { useFetch } from './service/hooks/useFetch';
 
 const App: React.FC = (props) => {
+  const [inputVal, setInputVal] = useState<string>('');
+  const [isSearching, setIsSearching] = useState<boolean>(false);
+  const [urlStr, setUrlStr] = useState<any>([]);
+
+  const menuContainerRef = useRef() as React.MutableRefObject<HTMLDivElement>;
+
+  const isLoading = useSelector( (state: RootState)  => state.isLoading);
+  const isOpen = useSelector( (state: RootState) => state.menuOpen);
+  const isMobile = useSelector( (state: RootState) => state.isMobile);
+  const dispatch = useDispatch();
+
+  const mobileMenuTabStyle = (
+    isOpen ? 'translateY(-54vh) translateX(50%)' : 'translateY(-18vh) translateX(50%)'
+  );
+  const desktopMenuTabStyle = (
+    isOpen ? 'translateX(-24vw)' : 'translateX(0)'
+  );
+
+  // const data = useFetch("https://raw.githubusercontent.com/carbonmap/ccm-front-end/master/dummy_data/reporting_entities/index.json");
+
+  // useEffect(() => {
+  //   if(data !== null) {
+  //     setUrlStr(data)
+  //     console.log(data)
+  //   }    
+  // },[data]);
+
+  useEffect(() => {
+
+    handleWindowSizeChange(dispatch);
+
+    window.addEventListener('resize', () => handleWindowSizeChange(dispatch));
+    return () => {
+        window.removeEventListener('resize', () => handleWindowSizeChange(dispatch));
+    }
+
+  },[]);
+
   return (
-      <IonApp>
-        <IonHeader>
-          <Toolbar />
-        </IonHeader>
-        <SearchBar />
-        <IonContent>
-          <SideMenu />
-          <Map />
-        </IonContent>
-      </IonApp>
+    <IonApp style={{ width: '100vw' }}>
+
+      {urlStr.length > 0 ?
+          <div className="spinner-container">
+            <Spinner /> 
+          </div>
+         : 
+         <IonApp style={{ width: '100%', height: '100%' }}>
```

Comment:
Interested in this IonApp within another IonApp - I haven't seen that construct in the samples I've seen. How does this work?